registration/forms.py

```diff
@@ -355,6 +391,32 @@ def update_user(self, application, commit=True):
                 application.user.profile.save()
                 application.save()
 
+        if self.restart_fields_changed():
+            messages.add_message(self.request, messages.INFO,
+                                 "Thank you for submitting these changes - we will rebuild your instance to "
+                                 "apply them, and email you to confirm once it is up to date.")
+
+            # Recreate the ansible variables in the instance
+            instance = application.instance
+            if instance:
+                instance.configuration_extra_settings = get_design_fields_as_yaml(application)
```

Comment:
Simple theme configuration variables should not be stored in `configuration_extra_settings`. The `configuration_extra_settings` field should only be used for settings which are entered manually and are not automated by Ocim.

Add a new field `configuration_theme_settings` instead. You can check how this is done for storage settings, which are stored in `configuration_storage_settings` and most related code is contained in the [openedx_storage.py mixin](https://github.com/open-craft/opencraft/blob/0afe1e19eeb1c6fc2b338294ba1c1cbddd64b365/instance/models/mixins/openedx_storage.py). The value [get set whenever creating a new appserver](https://github.com/open-craft/opencraft/blob/0afe1e19eeb1c6fc2b338294ba1c1cbddd64b365/instance/models/openedx_instance.py#L228). If you grep the code for `configuration_storage_settings`, you will see how it all fits together. Please follow the same approach for `configuration_theme_settings`.